src/clj/game/cards/ice.clj

```diff
@@ -1456,6 +1456,33 @@
                    :prompt "Choose an ICE to install from HQ"
                    :effect (req (corp-install state side target (zone->name (first (:server run))) {:no-install-cost true}))}]}
 
+   "Formicary"
+   {:req (req (and (:run @state)
+                   (zero? (:position run))
+                   (not (contains? run :corp-phase-43))
+                   (not (contains? run :successful))))
+    :effect (effect (resolve-ability
+                     {:optional
+                      {:prompt "Move Formicary?"
+                       :yes-ability {:msg "rez and move Formicary. The Runner is now approaching Formicary."
+                                     :effect (req
+                                              (move state side card
+                                                    [:servers (first (:server run)) :ices]
+                                                    {:front true})
+                                              (swap! state assoc-in [:run :position] 1))}
+                       :no-ability {:msg "rez Formicary without moving it"}}}
+                     card nil))
+    :subroutines [{:label "End the run unless the Runner suffers 2 net damage"
+                   :effect (req (wait-for (resolve-ability
```

Comment:
Subroutines currently don't plug into the async system, but to future proof, the sub should be `:async true` in order to use `wait-for`.